plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -83,14 +118,17 @@ export class TaskWorker {
     return new TaskWorker({
       taskBroker: taskBroker,
       runners: { workflowRunner },
+      concurrentTasksLimit,
     });
   }
 
   start() {
     (async () => {
       for (;;) {
+        await this.waitWorkerToBeAvailable();
```

Comment:
Could we use something pre-existing like [p-queue](https://github.com/sindresorhus/p-queue), plimit or alike so that we don't have to maintain our own locking.

Reply:
Awesome. Thanks for the suggestions. Totally agree about using a community-backed solution. I briefly looked into p-limit, I don't think we can used that, because it needs the array of promises before the creation of the limiting mechanism. `p-queue` looks to be more inline with the existing software architecture. I can look more into implementing it this week.

Reply:
I implemented a taskQueue using the p-queue library instead of creating own locking mechanism